### PR TITLE
Complete "Re-Creating Python's `async def`/coroutines in Rust"

### PR DESCRIPTION
**Pull request by ThibaultLemaire**
with a working example of python code calling a rust-made coroutine

### CODE REVIEW

**Comment by ThibaultLemaire** [comment]
Hi! Thanks for this example. I was able to finish it off by myself, so I thought why not share my code?

File: README.md [hunk 1]
```diff
@@ -391,13 +395,43 @@ impl PyIterProtocol for MyCoroutine {
     }
 }
 
+// Exposing our custom awaitable to Python.
+// This will behave like a coroutine.
+#[pyfunction]
+fn my_coroutine() -> PyResult<MyCoroutine> {
+    Ok(MyCoroutine {})
+}
```
Comment (ThibaultLemaire) [inline README.md:403]: Either that or we could expose a constructor for `MyCoroutine`. I thought this was more in line with the spirit of the previous examples, but it's your call.

File: README.md [hunk 2]
```diff
@@ -391,13 +395,43 @@ impl PyIterProtocol for MyCoroutine {
...
+    print(f"my_coroutine returned with: {result!r}"
+
+# But note that this won't work:
+# asyncio.run(await_from_rust.my_coroutine())
+# because asyncio won't recognise it as a `coroutine` object
+
+asyncio.run(main())
+```
+
+### Using Rust's futures as Python awaitables
```
Comment (ThibaultLemaire) [inline README.md:435]: I'm just teasing a bit :wink:  This is what I'd like to be able to do in the end. I can remove it if that's not where you want to take this examples page though.

**Comment by ChillFish8** [comment]
Firstly, thanks this is pretty awesome to have some more people contribute to this! Haven't had too much time recently to be able to work on it so having others helps a lot!
There are some tweaks i think that might be worth changing before i merge it though, ill comment them.

File: README.md (hunk 1 above)
Comment (ChillFish8) [reply README.md:403]: I think it would be a good idea to remove the `PyResult<>` in this case it isn't needed so probably best to keep it simple.

Comment (ThibaultLemaire) [reply README.md:403]: Didn't know you could do that. Even cleaner that way! Commit amended

**Pull request merged**